main.go

```diff
@@ -37,50 +36,71 @@ func getConsulClient(httpAddr string, token string) *api.Client {
 	return consulClient
 }
 
-func registerConsulService(name string, frequency int, client *api.Client) {
-	fmt.Printf("Registering '%s' in consul\n", name)
-	client.Agent().ServiceRegister(&api.AgentServiceRegistration{
+func registerConsulService(name string, frequency time.Duration, client *api.Client) {
+	log.Printf("[ConsulWrapper] Registering '%s' in consul\n", name)
+	err := client.Agent().ServiceRegister(&api.AgentServiceRegistration{
 		ID:   name,
 		Name: name,
 		Check: &api.AgentServiceCheck{
 			CheckID:                        name,
 			Name:                           name,
-			DeregisterCriticalServiceAfter: strconv.Itoa(2*frequency) + "s",
-			TTL:                            strconv.Itoa(3*frequency) + "s",
+			DeregisterCriticalServiceAfter: fmt.Sprintf("%d", int64(2.1*frequency.Seconds())) + "s",
+			TTL:                            fmt.Sprintf("%d", int64(3.1*frequency.Seconds())) + "s",
 		},
 	})
+	if err != nil {
+		log.Fatal(err)
+	}
 }
 
 func deregisterConsulService(name string, client *api.Client) {
-	fmt.Printf("Deregister '%s' from consul\n", name)
-	client.Agent().ServiceDeregister(name)
+	log.Printf("[ConsulWrapper] Deregistering '%s' from consul\n", name)
+	err := client.Agent().ServiceDeregister(name)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "Usage of %s:\n\n", os.Args[0])
+	fmt.Fprintf(os.Stderr, "\t%s [options] -service <ServiceName> -command <Command> [Args]\n\n", os.Args[0])
+	flag.PrintDefaults()
 }
 
 func main() {
 	tokenPrt := flag.String("token", "", "Consul token used for registration")
 	servicePtr := flag.String("service", "", "Consul Service Name")
-	checkFrequencyPtr := flag.Int("frequency", 30, "Health Check Frequency (in seconds)")
+	checkFrequencyPtr := flag.Duration("frequency", time.Duration(30), "Health Check Frequency (in seconds)")
 	commandPtr := flag.String("command", "", "Command to run")
```

Comment:
You can remove this and just use `flag.Args()`

Reply:
Done